cpp/src/io/orc/writer_impl.cu

```diff
@@ -1937,6 +1944,91 @@ string_dictionaries allocate_dictionaries(orc_table_view const& orc_table,
           std::move(is_dict_enabled)};
 }
 
+struct string_length_functor {
+  __device__ inline size_type operator()(int const i) const
+  {
+    // we translate from 0 -> num_chunks * 2 because each statistic has a min and max
+    // string and we need to calculate lengths for both.
+    if (i >= num_chunks * 2) return 0;
+
+    // min strings are even values, max strings are odd values of i
+    auto const min = i % 2 == 0;
```

Comment:
Darn, this one's hard to name well.
Maybe something like `should_copy_min`/`is_assigned_to_copy_min` ?

Reply:
yeah, I struggled with a name as well. I decided to keep it short because it doesn't have much of a lifetime, but it's been brought up multiple times, so this seems warranted.